src/lib/query/sriov.rs

```diff
@@ -178,12 +186,34 @@ fn get_vf_iface_name(pf_name: &str, sriov_id: &u32) -> Option<String> {
     read_folder(&sysfs_path).pop()
 }
 
+fn is_drivers_autoprobe(pf_name: &str) -> Option<bool> {
+    let sysfs_path =
+        format!("/sys/class/net/{pf_name}/device/sriov_drivers_autoprobe");
+    match std::fs::read_to_string(sysfs_path) {
+        Ok(s) => match s.trim().parse::<u8>() {
+            Ok(u) => Some(u == 1),
+            _ => None,
+        },
+        _ => None,
+    }
+}
+
+fn get_num_vfs(pf_name: &str) -> Option<u32> {
+    let sysfs_path = format!("/sys/class/net/{pf_name}/device/sriov_numvfs");
+    match std::fs::read_to_string(sysfs_path) {
+        Ok(s) => match s.trim().parse::<u32>() {
+            Ok(u) => Some(u),
+            _ => None,
+        },
+        _ => None,
+    }
+}
+
 // SR-IOV can be disabled on BIOS but the VFs netlink attribute will be there. In order to
 // understand if the SR-IOV can be configured we must look at PCI level. If sriov_numvfs is present
 // we can assume that the NIC is SR-IOV capable and it is enabled.
 pub(crate) fn sriov_is_enabled(pf_name: &str) -> bool {
-    let sysfs_path = format!("/sys/class/net/{pf_name}/device/sriov_numvfs");
-    std::fs::File::open(sysfs_path).is_ok()
+    get_num_vfs(pf_name).is_some()
```

Comment:
I would remove this change. There is no need to read the file to check its existance.